scala/scala-impl/src/org/jetbrains/plugins/scala/project/Versions.scala

```diff
@@ -55,7 +56,7 @@ object Versions  {
   private def loadLinesFrom(url: String): Try[Seq[String]] = {
     Try(HttpConfigurable.getInstance().openHttpConnection(url)).map { connection =>
       try {
-        if(url.contains(Entity.Hydra.url))
+        if(url.contains(HydraApplicationSettings.getInstance().getHydraRepositoryUrl))
```

Comment:
I want to put the Authorization Request property only if the versions are downloaded from the Hydra Repo. Otherwise I don't want to pass that to the connection.

Reply:
I see, this code path can be hit also when not downloading Hydra. We should refactor this to something like:

```
private def loadLinesFrom(url: URL)(prepareConnection: <HttpConnectionType> => Unit): Try[Seq[String]] = {
  Try(HttpConfigurable.getInstance().openHttpConnection(url)).map { connection =>
    try {
      prepareConnection(connection)
      Source.fromInputStream(connection.getInputStream).getLines().toVector
    } finally {
      connection.disconnect()
    }
  }
}
```

And when calling `loadLinesFrom` for the Hydra Entity, we should do

```
loadLinesFrom(<HydraURL>) { connection =>
  connection.setRequestProperty("Authorization", "Basic " + HydraCredentialsManager.getBasicAuthEncoding())
}
```

Reply:
I don't know if is that easy to refactor because the `loadLinesFrom` method is called from 
`private def loadVersionsFrom(url: String, filter: PartialFunction[String, String])`